api/types/plugin.go

```diff
@@ -71,6 +71,10 @@ type PluginConfig struct {
 	// Required: true
 	Network PluginConfigNetwork `json:"Network"`
 
+	// propagated mount
+	// Required: true
+	PropagatedMount string `json:"PropagatedMount"`
```

Comment:
https://github.com/docker/docker/blob/master/docs/extend/config.md needs to be updated. Please do that alongside.